test/code/providers/Container_HostInventory_Class_Provider_UnitTest.cpp

```diff
@@ -0,0 +1,85 @@
+#include <set>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string>
+#include <unistd.h>
+#include <uuid/uuid.h>
+#include <vector>
+
+#include <scxcorelib/scxcmn.h>
+#include <scxcorelib/scxprocess.h>
+#include <scxcorelib/stringaid.h>
+#include <testutils/scxunit.h>
+#include <testutils/providertestutils.h>
+
+#include "Container_HostInventory_Class_Provider.h"
+#include "cjson/cJSON.h"
+
+using namespace std;
+using namespace SCXCoreLib;
+
+class ContainerHostInventoryTest : public CppUnit::TestFixture
+{
+    CPPUNIT_TEST_SUITE(ContainerHostInventoryTest);
+    CPPUNIT_TEST(TestEnumerateInstances);
+    CPPUNIT_TEST_SUITE_END();
+
+public:
+    vector<string> containers;
+
+    static string NewGuid()
+    {
+        uuid_t uuid;
+        uuid_generate_random(uuid);
+        char s[37];
+        uuid_unparse(uuid, s);
+        return s;
+    }
+
+    static string RunCommand(const char* command)
+    {
+        istringstream processInput;
+        ostringstream processOutput;
+        ostringstream processErr;
+
+        CPPUNIT_ASSERT(!SCXProcess::Run(StrFromMultibyte(string(command)), processInput, processOutput, processErr, 0));
+        CPPUNIT_ASSERT_EQUAL(processErr.str(), string());
+
+        return processOutput.str();
+    }
+
+public:
+    
+protected:
+    void TestEnumerateInstances()
+    {
+        wstring errMsg;
+        TestableContext context;
+
+        vector<wstring> m_keyNames;
+        m_keyNames.push_back(L"InstanceID");
+
+        // Enumerate provider
+        StandardTestEnumerateInstances<mi::Container_HostInventory_Class_Provider>(m_keyNames, context, CALL_LOCATION(errMsg));
+
+        CPPUNIT_ASSERT(context.Size());
+		
+        // Only check that the values are present and within the valid range because it is not possible to create a controlled environment
+        for (unsigned i = 0; i < context.Size(); ++i)
+        {
+            wstring instanceId = context[i].GetProperty(L"InstanceID", CALL_LOCATION(errMsg)).GetValue_MIString(CALL_LOCATION(errMsg));
+            CPPUNIT_ASSERT(instanceId.length());
```

Comment:
This is a really weak test. You're testing that some non-empty string was returned, but that's it. Can't you do a little more robust checking, like that the values are actually reasonable?

Same issue for everything below, I think you can do a lot more robust checking. A crappy unit test can miss a lot of stuff. A good unit test will catch a lot of problems very early in the DEV cycle (a very good thing).